ext-dynamic-commitments.md

```diff
@@ -0,0 +1,423 @@
+# Extension Bolt ZZZ: Dynamic Commitments
+
+Authors:
+  * Keagan McClelland <keagan@lightning.engineering>
+  * Olaoluwa Osuntokun <roasbeef@lightning.engineering>
+  * Eugene Siegel <eugene@lightning.engineering>
+
+Created: TODO
+
+# Table of Contents
+
+TODO
+
+# Introduction
+
+## Abstract
+
+This document describes a protocol for changing channel parameters that were
+negotiated at the conception of the channel. Implementation of the protocol
+described in this document will enable channel peers to re-negotiate and update
+channel terms as if the channel was being opened for the first time while
+avoiding UTXO churn whenever possible and therefore preserving the continuity of
+identity for the channel whose terms are being changed.
+
+## Motivation
+
+It is well understood that closing channels is a costly thing to do. Not only is
+it costly from a chain fees perspective (where we pay for moving the funds from
+the channel UTXO back to the main wallet collection), it is also costly from a
+service availability and reputation perspective.
+
+After channels are closed, they are no longer usable for forwarding HTLC traffic
+and even if we were to immediately replace the channel with another equally
+capable one, the closure event is visible to the entire network. Since routes
+are computed by the source, the network-wide visibility of channel closures
+directly impacts whether or not the sender will be able to use a channel.
+
+Beyond that, one of the pathfinding heuristics that is frequently used to assess
+channel reliability is the channel age. The longevity of a channel is therefore
+a key asset that any running Lightning node should want to preserve, if
+possible.
+
+It follows from the above that we should try to minimize channel closure events
+when we can manage to do so. This is the main motivation of this proposal. Prior
+to this extension BOLT, there hasn't been a way to change some of the channel
+parameters established in the `{open|accept}_channel` messages without resorting
+to a full channel closure, even if the channel counterparty consents. This
+limitation can be remediated by introducing a protocol to renegotiate these
+parameters.
+
+## Preliminaries
+
+This proposal includes a detailed section on the preliminaries to document some
+of the rationale for the design that is presented later. If you are a Bitcoin
+and Lightning Network protocol expert or you are uninterested in the thought
+process behind what is presented here, you may wish to skip to the Design
+Overview section to save time.
+
+### Channel Opening Parameters
+
+As described in BOLT 2, during the channel opening procedure there are a number
+of parameters specified in the `open_channel` and `accept_channel` messages that
+remain static over the lifetime of the channel. A subset of these are updatable
+using other messages defined in the protocol. However, after accounting for the
```

Comment:
I am missing `htlc_minimum_msat`. 

In `open_channel`, it is the smallest amount a node will accept. In LND, it is defined by the config parameter `bitcoin.minhtlc`. After the channel is opened, the peer sees this value as a local constraint in `ListChannels`. This cannot be changed at the moment.

However, this is not the same as `htlc_minimum_msat`, which is defined in the `channel_update` message. This latter value represents the minimum amount we can send through the channel. It is constrained by the value mentioned above, as set by my peer.

Reply:
Good observation. I mistakenly thought that the one defined in the `channel_update` message was the same.